### PR TITLE
Implemented first task by adding new model, its repository class and …

### PR DESCRIPTION
**Pull request by f3ops3**
…service

### CODE REVIEW

File: src/main/java/springweb/courseproject/repository/BookRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Couldn't save a book = " + book, e);
```
Comment (Elena-Bruyako) [inline src/main/java/springweb/courseproject/repository/BookRepositoryImpl.java:34]:
```suggestion
            throw new DataProcessingException("Couldn't save a book = " + book, e);
```

File: src/main/java/springweb/courseproject/repository/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import java.util.List;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    private SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/springweb/courseproject/repository/BookRepositoryImpl.java:18]:
```suggestion
@Repository
@RequiredArgsConstructor
public class BookRepositoryImpl implements BookRepository {
    private final SessionFactory sessionFactory;
```

File: src/main/java/springweb/courseproject/repository/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Book", Book.class).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Couldn't find all books", e);
```
Comment (Elena-Bruyako) [inline src/main/java/springweb/courseproject/repository/BookRepositoryImpl.java:47]:
```suggestion
            throw new DataProcessingException("Couldn't find all books", e);
```

File: src/main/java/springweb/courseproject/service/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
+package springweb.courseproject.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import springweb.courseproject.model.Book;
+import springweb.courseproject.repository.BookRepository;
+import java.util.List;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/springweb/courseproject/service/BookServiceImpl.java:9]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/springweb/courseproject/service/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import springweb.courseproject.model.Book;
+import springweb.courseproject.repository.BookRepository;
+import java.util.List;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
```
Comment (Elena-Bruyako) [inline src/main/java/springweb/courseproject/service/BookServiceImpl.java:14]: remove constructor

File: src/main/java/springweb/courseproject/service/BookServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import springweb.courseproject.model.Book;
+import springweb.courseproject.repository.BookRepository;
+import java.util.List;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private BookRepository bookRepository;
```
Comment (Elena-Bruyako) [inline src/main/java/springweb/courseproject/service/BookServiceImpl.java:11]:
```suggestion
    private final BookRepository bookRepository;
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**